pypeit/coadd2d.py

```diff
@@ -169,39 +173,114 @@ def __init__(self, spec2d, spectrograph, par, det=1, offsets=None, weights='auto
         self.debug_offsets = debug_offsets
         self.debug = debug
+        self.offsets = None
         self.stack_dict = None
         self.pseudo_dict = None
 
         self.objid_bri = None
         self.slitid_bri  = None
         self.snr_bar_bri = None
+        self.use_weights = None
+        self.wave_grid = None
+        self.good_slits = None
+        self.maskdef_offset = None
 
 
         # Load the stack_dict
         self.stack_dict = self.load_coadd2d_stacks(self.spec2d)
         self.pypeline = self.spectrograph.pypeline
 
+        self.nexp = len(self.spec2d)
+
         # Check that there are the same number of slits on every exposure
         nslits_list = [slits.nslits for slits in self.stack_dict['slits_list']]
-        if not len(set(nslits_list))==1:
+        if not len(set(nslits_list)) == 1:
             msgs.error('Not all of your exposures have the same number of slits. Check your inputs')
-        # TODO: Do the same check above but for the shape and binning
-        # of the input images?
-        self.nslits = nslits_list[0]
-        self.nexp = len(self.stack_dict['specobjs_list'])
+        # This is the number of slits of the single (un-coadded) frames
+        self.nslits_single = nslits_list[0]
+
+        # Check that nspec is the same for all the exposures
+        nspec_list = [slits.nspec for slits in self.stack_dict['slits_list']]
+        if not len(set(nspec_list)) == 1:
+            msgs.error('Not all of your exposures have the same spectral dimension. Check your inputs')
         self.nspec = self.stack_dict['slits_list'][0].nspec
+
+        # Check that binning is the same for all the exposures
+        binspec_list = [slits.binspec for slits in self.stack_dict['slits_list']]
+        binspat_list = [slits.binspat for slits in self.stack_dict['slits_list']]
+        if not len(set(binspec_list)) == 1:
+            msgs.error('Not all of your exposures have the same spectral binning. Check your inputs')
+        if not len(set(binspat_list)) == 1:
+            msgs.error('Not all of your exposures have the same spatial binning. Check your inputs')
         self.binning = np.array([self.stack_dict['slits_list'][0].binspec,
                                  self.stack_dict['slits_list'][0].binspat])
+
         self.spat_ids = self.stack_dict['slits_list'][0].spat_id
 
         # If smoothing is not input, smooth by 10% of the spectral dimension
         self.sn_smooth_npix = sn_smooth_npix if sn_smooth_npix is not None else 0.1*self.nspec
 
-        # maskdef info
-        self.maskdef_id = np.array([slits.maskdef_id for slits in self.stack_dict['slits_list']])
-        self.maskdef_offset = np.array([slits.maskdef_offset for slits in self.stack_dict['slits_list']])
-        self.maskdef_objpos = np.array([slits.maskdef_objpos for slits in self.stack_dict['slits_list']])
-        self.maskdef_slitcen = np.array([slits.maskdef_slitcen for slits in self.stack_dict['slits_list']])
+    @property
```

Comment:
This seems pretty redundant. You are defining a property that just equal to an attribute of the class. Avoid state dependent properties. They just create a confusing control flow, but add no real value.